state/src/interface/raft.rs

```diff
@@ -0,0 +1,26 @@
+//! State interface methods for modifying the raft config state
+
+use common::types::gossip::WrappedPeerId;
+
+use crate::{
+    error::StateError, notifications::ProposalWaiter,
+    replication::network::address_translation::PeerIdTranslationMap, State, StateTransition,
+};
+
+impl State {
+    // -----------
+    // | Setters |
+    // -----------
+
+    /// Add a raft peer as a learner to the local cluster
+    pub fn add_raft_learner(&self, peer_id: WrappedPeerId) -> Result<ProposalWaiter, StateError> {
+        // Add the peer ID to the translation map
+        let mut locked_map = self.translation_map.write().expect("translation map poisoned");
+        locked_map.insert(peer_id);
+        let raft_id = PeerIdTranslationMap::get_raft_id(&peer_id);
+        drop(locked_map); // Unlock the map
```

Comment:
nit: we can drop the lock before we calculate the raft id, while i'm sure hashing is quick it's not in the critical section here right